### PR TITLE
DocumentBase.update should automatically refresh date_updated

### PR DESCRIPTION
**Pull request by wojcikstefan**

### CODE REVIEW

File: tests/__init__.py [hunk 1]
```diff
@@ -88,6 +91,64 @@ class B(Document):
...
+
+        doc.text = 'new'
+        doc.save()
+        doc.reload()
+
+        self.assertEqual(doc.date_created, last_date_created)
+        self.assertTrue(doc.date_updated > last_date_updated)
+        last_date_updated = doc.date_updated
+
+        time.sleep(0.001)  # make sure some time passes between the updates
```
Comment (tsx) [inline tests/__init__.py]: Maybe using freezefrog is better than sleeping?

Comment (thomasst) [inline tests/__init__.py]: I think sleep should be fine here. We just want to make sure we advance by 1ms.

File: flask_common/documents.py [hunk 2]
```diff
@@ -101,7 +107,16 @@ def delete(self, **kwargs):
         # delete only if already saved
         if self.pk:
             self.is_deleted = True
-            self.update(set__is_deleted=self.is_deleted)
+            update_kwargs = {
+                'set__is_deleted': self.is_deleted
+            }
+
+            # we don't want to update date_updated for deleted objects (in
+            # case they inherit from DocumentBase)
```
Comment (thomasst) [inline flask_common/documents.py]: Why not update `date_updated` to the date the document was deleted?

Comment (wojcikstefan) [inline flask_common/documents.py]: It was causing some issues with the close.io tests, but I dug deeper and realized it was just the test assumptions that were wrong.

**Pull request merged**